scapy/layers/dns.py

```diff
@@ -235,17 +235,16 @@ def i2h(self, pkt, x):
 class DNS(Packet):
     name = "DNS"
     fields_desc = [ ShortField("id", 0),
-                    BitField("qr", 0, 1),
-                    BitEnumField("opcode", 0, 4, {0:"QUERY",1:"IQUERY",2:"STATUS"}),
-                    BitField("aa", 0, 1),
-                    BitField("tc", 0, 1),
-                    BitField("rd", 0, 1),
-                    BitField("ra", 0, 1),
-                    BitField("z", 0, 1),
+                    BitEnumField("response", 0, 1, {0: "Query", 1: "Response"}),
+                    BitEnumField("opcode", 0, 4, {0: "QUERY",1: "IQUERY",2: "STATUS"}),
                     # AD and CD bits are defined in RFC 2535
-                    BitField("ad", 0, 1), # Authentic Data
-                    BitField("cd", 0, 1), # Checking Disabled
-                    BitEnumField("rcode", 0, 4, {0:"ok", 1:"format-error", 2:"server-failure", 3:"name-error", 4:"not-implemented", 5:"refused"}),
+                    FlagsField("flags", 0, 7,
```

Comment:
I am not quite found of this change either, as these fields are defined separately in the RFC. I think that it is better to be as close to RFCs as we could. This greatly ease Scapy usage.

Sorry =/